pkgs/by-name/yo/your_spotify/package.nix

```
@@ -0,0 +1,71 @@
+{
+  callPackage,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  lib,
+  makeWrapper,
+  mkYarnPackage,
+  nodejs,
+  prefetch-yarn-deps,
+  yarn,
+}: let
+  version = "1.8.0";
+  src = fetchFromGitHub {
+    owner = "Yooooomi";
+    repo = "your_spotify";
+    rev = "refs/tags/${version}";
+    hash = "sha256-umm7J5ADY2fl+tjs6Qeda5MX2P55u0eCqwW+DWLK8Kc=";
+  };
+  client = callPackage ./client.nix {inherit src version;};
+in
+  mkYarnPackage rec {
+    inherit version src;
+    name = "your_spotify-${version}";
+    offlineCache = fetchYarnDeps {
+      yarnLock = src + "/yarn.lock";
+      hash = "sha256-pj6owoEPx9gdtFvXF8E89A+Thhe/7m0+OJU6Ttc6ooA=";
+    };
+
+    configurePhase = ''
+      runHook preConfigure
+
+      export HOME=$(mktemp -d)
+      yarn config --offline set yarn-offline-mirror $offlineCache
+      fixup-yarn-lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+
+      runHook postConfigure
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+      pushd ./apps/server/
+      yarn --offline run build
+      popd
+      runHook postBuild
+    '';
+    nativeBuildInputs = [makeWrapper yarn prefetch-yarn-deps];
+    installPhase = ''
+      mkdir -p $out
+      cp -r node_modules $out/node_modules
```

```suggestion
      cp -r node_modules $out/share/your-spotify/node_modules
```

We should keep $out a bit clean

---

Done